TypingDNA ActiveLock integration (#1572)
DataDog/integrations-extras: typingdna_activelock/README.md

> 
> 

Reviewer: ```suggestion
```

# TypingDNA ActiveLock

## Overview

> [TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers. Once installed on a user's PC, it continuously verifies the user by the way they type. If an unauthorized typing pattern is detected, ActiveLock can instantly lock the computer and log the data to your desired logging platform (such as Datadog).

Reviewer: Can you provide more insight on the value of the integration with Datadog? For example, "With this integration, you can xyz"

Author: Added more context.

To visualize your data in Datadog, a custom ActiveLock app needs to be configured and installed. This is the same install for all of your company computers.

## Setup

### Configuration

To generate a Datadog API key:

1. Navigate to [Organization settings > API keys][4] in your Datadog account.
2. Click **+ New Key** to generate an API key.

To get your custom install app:

1. Complete [this custom install form][7] by submitting your newly generated API key and Datadog site region (such as US1 or EU), along with your company details.

> 2. You will then receive a custom ActiveLock app that you can install on your company computers and more information over email. This install has an initial limit of 10 seats, and comes with a default 30-day trial period. You will need to purchase a commercial ActiveLock license from TypingDNA in order to use it for more than 30 days, and/or on more than 10 computers. Contact [TypingDNA][2] for details or the reseller/partner through which you found us.

Reviewer: Can you provide more information on what will be provided "You will then receive a custom ActiveLock app that you can install on your company computers and more information over email." ?

Author: It is basically any installation information that may be needed. I updated that.

> 2. You will then receive a custom ActiveLock app that you can install on your company computers and more information over email. This install has an initial limit of 10 seats, and comes with a default 30-day trial period. You will need to purchase a commercial ActiveLock license from TypingDNA in order to use it for more than 30 days, and/or on more than 10 computers. Contact [TypingDNA][2] for details or the reseller/partner through which you found us.

Reviewer: ```suggestion
2. You will then receive a custom ActiveLock app that you can install on your company computers and more information over email.
```

Reviewer: Removing the additional information below since this integration will be listed on the integrations page so only mutual customers will be installing the integration.

3. After installation, your ActiveLock logs should start to appear in [Log Explorer][5].

> Note that if you came through a reseller/partner you should follow their instructions to get your custom install app (and commercial license).

Reviewer: Will customers of Datadog and TypingDNA be from resellers as well?

Author: TypingDNA ActiveLock can be sold by resellers in which case, the same integration will be installed, BUT the reseller will be able to manage the license, changing it from trial to pro, add seats and expiration. Usually the reseller will provide all details separately to the client.

# TypingDNA ActiveLock

## Overview

> [TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers. Once installed on a user's PC (MacOS or Windows), it continuously verifies the user by the way they type. If an unauthorized typing pattern is detected, ActiveLock can instantly lock the computer and log the data to your desired logging platform (such as Datadog).

Reviewer: ```suggestion
[TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers by detecting typing patterns and locking computers to protect sensitive data.
```

[TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers. Once installed on a user's PC, it continuously verifies the user by the way they type. If an unauthorized typing pattern is detected, ActiveLock can instantly lock the computer and log the data to your desired logging platform (such as Datadog).

To visualize your data in Datadog, a custom ActiveLock app needs to be configured and installed. This is the same install for all of your company computers.

## Setup

### Configuration

To generate a Datadog API key:

1. Navigate to [Organization settings > API keys][4] in your Datadog account.
2. Click **+ New Key** to generate an API key.

To get your custom install app:

1. Complete [this custom install form][7] by submitting your newly generated API key and Datadog site region (such as US1 or EU), along with your company details.
2. You will then receive a custom ActiveLock app that you can install on your company computers and more information over email.

> 3. After installation, your ActiveLock logs should start to appear in [Log Explorer][5].

Reviewer: Just a reminder to update the instructions once you have built out the page and form for customers to submit their api key.

Author: Thanks, I will, currently waiting for the page to be published, may take a few more days.

> 1. Complete [this custom install form][7] by submitting your newly generated API key and Datadog site region (such as US1 or EU), along with your company details.

Reviewer: ```suggestion
1. Complete [this custom install form][7] by submitting your newly generated API key, [Datadog site][9], and your company details.
```

### Validation

To view your ActiveLock logs in Datadog, navigate to the [Log Explorer][5] and enter `source:typingdna_activelock` in the search query.

To access the dashboard, navigate to the [Dashboard List][6] and search for the **TypingDNA ActiveLock** dashboard.

## Data Collected

### Log collection

TypingDNA ActiveLock logs are collected and sent to Datadog directly from each application.

> In order to view the logs correctly in the TypingDNA ActiveLock dashboard, you need to set the facets and measures in the [Configuration](#configuration) section.
> 

Reviewer: ```suggestion
```

> 2. You will then receive a custom ActiveLock app that you can install on your company computers along with installation instructions over email. This install has an initial limit of 10 seats, and comes with a default 30-day trial period. To remove trial limitations you need a full commercial license. If you don't have a commercial license already, contact [TypingDNA][2] for licensing or the reseller/partner through which you found us.
> 3. After installation, your ActiveLock logs should start to appear in [Log Explorer][5].

Reviewer: ```suggestion
2. You will receive an email including a custom ActiveLock app to install on your company computers, and installation instructions. 
	a. This install has an initial limit of 10 seats, and comes with a default 30-day trial period. To remove trial limitations you need a full commercial license. If you don't have a commercial license already, contact [TypingDNA][2] for licensing or the reseller/partner through which you found us.
3. After installation, your ActiveLock logs should start to appear in [Log Explorer][5].
```

> Note that if you came through a reseller/partner you should follow their instructions to get your custom install app (and commercial license).

Reviewer: ```suggestion
Note: If you are through a reseller/partner, please follow their instructions to get your custom install app (and commercial license).
```

## Troubleshooting

Need help? Contact [Datadog][1] or [TypingDNA support][2].

> [1]: https://docs.datadoghq.com/help/
> [2]: https://www.typingdna.com/contact
> [3]: https://www.typingdna.com/activelock
> [4]: https://app.datadoghq.com/organization-settings/api-keys
> [5]: https://app.datadoghq.com/logs
> [6]: https://app.datadoghq.com/dashboard/lists
> [7]: https://www.typingdna.com/clients/altrial
> [8]: https://app.datadoghq.com/integrations/typingdna_activelock

Reviewer: ```suggestion
[1]: https://docs.datadoghq.com/help/
[2]: https://www.typingdna.com/contact
[3]: https://www.typingdna.com/activelock
[4]: https://app.datadoghq.com/organization-settings/api-keys
[5]: https://app.datadoghq.com/logs
[6]: https://app.datadoghq.com/dashboard/lists
[7]: https://www.typingdna.com/clients/altrial
[8]: https://app.datadoghq.com/integrations/typingdna_activelock
[9]: https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site
```

# TypingDNA ActiveLock

## Overview

> [TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers. Once installed on a user's PC, it continuously verifies the user by the way they type. If an unauthorized typing pattern is detected, ActiveLock instantly locks the computer and/or logs the data to your desired logging platform (i.e. Datadog).

Author: logging the data is optional, "and" means it's by default

To visualize your data in Datadog, a custom ActiveLock app will need to be configured and installed. Note: This will be the same install for all of your company computers.

Configuration is straightforward and takes few minutes, please follow the steps in the **Configure** tab to get started.

## Setup

### Configuration

> 

Author: I just removed this because your suggestion includes the instruction in the actual place where it takes you, making it useless at this point. If it was in the previous tab (Overview it would have made some sense..)

> **I. Generate a Datadog API key**
> 1. Within your Datadog account, navigate to [Organization settings > API keys][4].
> 2. Generate a new "API key".
> 
> **II. Get your custom install app.**
> 1. Navigate to [this custom install form][7] and submit your newly generated API key and your Datadog Region (e.g. US1, EU) along with your company details.
> 2. Once we receive your information, we'll send you a custom ActiveLock app that you'll have to install on your company computers. We'll send any further details over email.
> 3. After installation, logs should start appearing in [Log explorer][5].
> 
> **III. Set up log features (facets/measures).**
> 1. Once logs flow into Datadog, Navigate to [Log explorer][5].
> 2. On the left side panel, select the "+Add" button to add the following facets/measures.
> Note: Facets and measures are case sensitive. Please include `@` in the path definition.

Reviewer 2: ```suggestion

To generate a Datadog API key:

1. Navigate to [**Organization settings** > **API keys**][4] in your Datadog account.
2. Click **+ New Key** to generate an API key.

To get your custom install app:

1. Complete [this custom install form][7] by submitting your newly generated API key and Datadog site region (such as US1 or EU), along with your company details.
2. Once we receive your information, you will receive a custom ActiveLock app that you can install on your company computers and more information details over email.
3. After installation, your ActiveLock logs should start to appear in the [Log Explorer][5].

To set up log features such as facets and measures:

1. Once logs are flowing into Datadog, navigate to the [Log Explorer][5].
2. On the left side panel, click **+Add** to add the following facets and measures. Facets and measures are case sensitive. Include `@` in the path definition.
```

> |Path| Facet/Measure |Type|Display name|

Reviewer 2: ```suggestion
|Path| Facet or Measure |Type|Display name|
```

|--|--|--|--|
|`@username`|facet|string|username|
|`@installID`|facet|string|installID|
|`@main_app`|facet|string|main_app|
|`@result`|facet|string|result|
|`@keyboard_id`|facet|string|keyboard_id|
|`@version`|facet|string|version|
| `@score` | measure |integer|score|
| `@speed` | measure |integer|speed|
| `@training_strength` | measure |integer|training_strength|

### Validation

> To view your ActiveLock logs in Datadog, navigate to [Log explorer][5] and set `source:typingdna_activelock`

Reviewer 2: ```suggestion
To view your ActiveLock logs in Datadog, navigate to the [Log Explorer][5] and enter `source:typingdna_activelock` in the search query.
```

> To open the dashboard, navigate to [All Dashboards][6] where you should find the **TypingDNA ActiveLock** dashboard.
> 

Reviewer 2: ```suggestion
To access the dashboard, navigate to the [Dashboard List][6] and search for the **TypingDNA ActiveLock** dashboard.

```

## Data Collected

### Log collection

> TypingDNA ActiveLock logs are collected and sent to Datadog directly from each app.
> 
> In order to be able to view the logs correctly in the TypingDNA ActiveLock dashboard you will need to set the facets and measures exactly as in the Configuration steps.

Reviewer 2: ```suggestion
TypingDNA ActiveLock logs are collected and sent to Datadog directly from each application.

In order to view the logs correctly in the TypingDNA ActiveLock dashboard, you need to set the facets and measures in the [Configuration](#configuration) section.

### Metrics

The TypingDNA ActiveLock integration does not include any metrics.

### Events

The TypingDNA ActiveLock integration does not include any events.

### Service Checks

The TypingDNA ActiveLock integration does not include any service checks.

```

Author: emarsha94 suggested to delete those 3 sentences before now you're asking me to add them back in, IMO you guys need to make sure you suggest the same things

## Troubleshooting

> Need help? Contact [Datadog support][1].
> Contact [TypingDNA support][2]

Reviewer 2: ```suggestion
Need help? Contact [Datadog][1] or [TypingDNA support][2].
```

[1]: https://docs.datadoghq.com/help/
[2]: https://www.typingdna.com/contact
[3]: https://www.typingdna.com/activelock
[4]: https://app.datadoghq.com/organization-settings/api-keys
[5]: https://app.datadoghq.com/logs
[6]: https://app.datadoghq.com/dashboard/lists

> [7]: https://forms.gle/3U9KxF7ySThVLDJg8

Author: I added the link as suggested but it doesn not make sense to reference it from itself, so I left it only in the link declaration at the end if we'll use it forward

# TypingDNA ActiveLock

## Overview

[TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers. Once installed on a user's PC, it continuously verifies the user by the way they type. If an unauthorized typing pattern is detected, ActiveLock instantly locks the computer and/or logs the data to your desired logging platform (i.e. Datadog).

> In order to get your data logged, analyzed and visualized in Datadog, you will need a custom ActiveLock install which will be configured to send the data directly to your Datadog. Once done, you'll use the same install for all your company computers.

Author: Partly inaccurate

Configuration is straightforward and takes few minutes, please follow the steps in the *Configure* tab to get started.

## Setup

### Configuration

> There are 3 main parts of the setup.

Author: I want to keep this because the 3 steps are done at separate times, and the user needs to understand this IMO

**1. Generate a Datadog API key**
Within your Datadog account, go to [Organization settings > API keys][4] and generate a new key.

**2. Get your custom install app.**
Send your newly generated API key AND your Datadog Region (e.g. US1, EU), to your TypingDNA account manager, and/or to datadog.support@typingdna.com. Once we receive your API key and Region, we'll send you a custom install app that you'll have to install on your company computers. New logs will start appearing in [Log explorer][5].

**3. Set up log features (facets/measures).**

> Now, once your data starts to log in Datadog, for ActiveLock dashboard to work correctly you'll need to set up a few more things. From [Log explorer][5], you'll need to add the following facets/measures (case sensitive) from the left side panel, *+Add* button. It is important to use `@` in the path definition.

Author: I made most changes as you asked, but I need just 3 steps, can't have 3 steps with 3 substeps each, it's too confusing.

# TypingDNA ActiveLock

## Overview

[TypingDNA ActiveLock][3] is a Continuous Endpoint Authentication app that helps prevent unauthorized access to your company computers. Once installed on a user's PC, it continuously verifies the user by the way they type. If an unauthorized typing pattern is detected, ActiveLock can instantly lock the computer and log the data to your desired logging platform (such as Datadog).

To visualize your data in Datadog, a custom ActiveLock app needs to be configured and installed. This is the same install for all of your company computers.

## Setup

### Configuration

To generate a Datadog API key:

1. Navigate to [Organization settings > API keys][4] in your Datadog account.
2. Click **+ New Key** to generate an API key.

To get your custom install app:

1. Complete [this custom install form][7] by submitting your newly generated API key and Datadog site region (such as US1 or EU), along with your company details.
2. You will then receive a custom ActiveLock app that you can install on your company computers and more information over email.
3. After installation, your ActiveLock logs should start to appear in [Log Explorer][5].

> To set up log features such as facets and measures:
> 
> 1. Once logs are flowing into Datadog, navigate to [Log Explorer][5].
> 2. On the left side panel, click **+Add** to add the following facets and measures. Facets and measures are case sensitive. Include `@` in the path definition.
> 
> |Path| Facet or Measure |Type|Display name|
> |--|--|--|--|
> |`@username`|facet|string|username|
> |`@installID`|facet|string|installID|
> |`@main_app`|facet|string|main_app|
> |`@result`|facet|string|result|
> |`@keyboard_id`|facet|string|keyboard_id|
> |`@version`|facet|string|version|
> | `@score` | measure |integer|score|
> | `@speed` | measure |integer|speed|
> | `@training_strength` | measure |integer|training_strength|

Reviewer: ```suggestion
```

### Validation

To view your ActiveLock logs in Datadog, navigate to the [Log Explorer][5] and enter `source:typingdna_activelock` in the search query.

To access the dashboard, navigate to the [Dashboard List][6] and search for the **TypingDNA ActiveLock** dashboard.

## Data Collected

### Log collection

TypingDNA ActiveLock logs are collected and sent to Datadog directly from each application.

In order to view the logs correctly in the TypingDNA ActiveLock dashboard, you need to set the facets and measures in the [Configuration](#configuration) section.

> ### Metrics
> 
> The TypingDNA ActiveLock integration does not include any metrics.
> 
> ### Events
> 
> The TypingDNA ActiveLock integration does not include any events.
> 
> ### Service Checks
> 
> The TypingDNA ActiveLock integration does not include any service checks.
> 
> 

Reviewer: ```suggestion
```
